src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/hearingpart/AssignHearingPartToSessionAction.java

```diff
@@ -1,107 +1,139 @@
+    public void getAndValidateEntities() {
+        hearingPart = hearingPartRepository.findOne(hearingPartId);
+        targetSession = sessionRepository.findOne(relationship.getSessionData().getSessionId());
+
+        if (targetSession == null) {
+            throw new RuntimeException("Target sessions cannot be null!");
+        } else if (hearingPart == null) {
+            throw new RuntimeException("Hearing part cannot be null!");
+        }
+    }
+
+    @Override
+    public UUID[] getAssociatedEntitiesIds() {
+        final List<UUID> entitiesIds = Arrays.asList(
+            hearingPart.getId(), hearingPart.getSessionId(), hearingPart.getHearingId(), targetSession.getId()
+        );
+        return entitiesIds.toArray(new UUID[0]);
+    }
+
+    @Override
+    public void act() {
+        // is this missing:? entityManager.detach(hearingPart);
+        try {
+            previousHearingPart = objectMapper.writeValueAsString(hearingPart);
+            previousHearing = objectMapper.writeValueAsString(hearingPart.getHearing());
+            previousSession = hearingPart.getSession();
+            hearingPart.setVersion(relationship.getHearingPartVersion());
+
+            UUID targetSessionId = (targetSession == null) ? null : targetSession.getId();
+            hearingPart.setSessionId(targetSessionId);
+            hearingPart.setSession(targetSession);
+        } catch (JsonProcessingException e) {
+            throw new SnlEventsException(e);
+        }
+    }
+
+    @Override //Done although hearing and session for user transactionDAta are not needed
+    public List<UserTransactionData> generateUserTransactionData() {
+        HearingPart savedHearingPart = hearingPartRepository.save(hearingPart);
+
+        List<UserTransactionData> userTransactionDataList = new ArrayList<>();
+
+        userTransactionDataList.add(new UserTransactionData("hearing",
+            savedHearingPart.getHearingId(),
+            previousHearing,
+            "update",
+            "update",
+            0)
+        );
+
+        userTransactionDataList.add(new UserTransactionData("hearingPart",
+            savedHearingPart.getId(),
+            previousHearingPart,
+            "update",
+            "update",
+            1)
+        );
+
+        if (previousSession != null) {
+            userTransactionDataList.add(getLockedSessionTransactionData(previousSession.getId()));
+        }
+        userTransactionDataList.add(getLockedSessionTransactionData(targetSession.getId()));
+
+        return userTransactionDataList;
+    }
+
+    @Override
+    public List<FactMessage> generateFactMessages() {
+        String msg;
+        try {
+            msg = factsMapper.mapHearingPartToRuleJsonMessage(hearingPart);
+        } catch (com.fasterxml.jackson.core.JsonProcessingException e) {
+            throw new SnlEventsException(e);
+        }
+        return Collections.singletonList(new FactMessage(RulesService.UPSERT_HEARING_PART, msg));
```

Comment:
usually we use Arrays.asList()

Reply:
for single entry, not necessarily + IDE is complaining, in the end, it doesn't matter much.